src/App.js

```diff
@@ -48,13 +51,17 @@ function App() {
 
   // handle collection selector
   function handleCollectionChange(event) {
-    const selectedCollection = sortedCollections[event.target.value];
-    setCurrentCollection(selectedCollection);
+    setCurrentCollectionId(event.target.value);
   }
 
-  // function updateItemPrice() {
-  //   // update item price
-  // }
+  function updateItemPrice(collectionId, itemId, price) {
+    const nextItems = collections[collectionId].items.map((item, index) => {
+      return index === itemId ? { ...item, price: parseFloat(price) } : item;
+    });
+    const nextCollection = { ...collections[collectionId], items: nextItems }
+    const nextCollections = { ...collections, [collectionId]: nextCollection }
+    setCollections(nextCollections);
+  }
```

Comment:
Could you elaborate a bit on what you're wanting to accomplish here? Is this just reconstructing the data structure with the new updates?

Reply:
Yes, this is how I'm applying the change in price to the `collections` state -- but as you may have noticed, doing it this way re-renders the entire Product Grid after every keystroke, losing focus on the price input you're editing.

This is the bug I mentioned, so I'm rethinking how and when I would need to rebuild the `collections` data.

Reply:
[d14b452](https://github.com/bengoshow/unit-converter/pull/2/commits/d14b4527eec2e16cd49cd8839b11c73842a6c481) moves the entire process of storing the price change data to the collections state. Now instead of storing the price change after every input, it only updates the collection if the user selects a different collection from the dropdown.